pkg/profile/artifact_test.go

```diff
@@ -611,120 +622,11 @@ status: {}
 						GitRepoName:      gitRepoName,
 					})
 					err := maker.Make(pSub)
-					Expect(err).To(MatchError(ContainSubstring("failed to generate resources for nested profile \"bitnami-nginx\": failed to generate resources for nested profile \"bitnami-nginx/recursive\": failed to generate resources for nested profile \"bitnami-nginx/recursive/recursive\": recursive artifact detected: profile https://github.com/org/repo-name-nested on branch")))
+					Expect(err).To(MatchError(ContainSubstring("recursive artifact detected: profile example.com/nested on branch main contains an artifact that points recursively back at itself")))
 				})
 			})
 		})
 	})
-	When("there is a single profile repository", func() {
```

Comment:
this file is still 600 lines?!!?! Oh cmon! :sob:

Reply:
Yeah I know. :/ it's mostly setup now... and errors. I don't know what to do with it! :( I could put every test into separate files. :D I don't know. :/

Reply:
isn't the majority of this file duplicating testing the other builder tests? I think a lot of it could get deleted, and replaced with `fake` builders to reduce the cost and make it easier to make changes to the builders without having to change these test. WDYT?

Reply:
I.... I think so? But not the file generation which we also want to test, right? So I'm super on the fence how to handle this. :D